djangocms_picture/cms_plugins.py

```diff
@@ -1,46 +1,67 @@
 # -*- coding: utf-8 -*-
-
-from __future__ import unicode_literals
-
-try:
-    import urlparse
-except ImportError:
-    from urllib import parse as urlparse
-
-from django.conf import settings
 from django.utils.translation import ugettext_lazy as _
 
 from cms.plugin_base import CMSPluginBase
 from cms.plugin_pool import plugin_pool
 
-from .models import Picture
+from . import models
+from . import forms
 
 
 class PicturePlugin(CMSPluginBase):
-    model = Picture
-    name = _("Picture")
-    render_template = "cms/plugins/picture.html"
+    model = models.Picture
+    form = forms.PictureForm
+    name = _('Image')
+    render_template = 'djangocms_picture/picture.html'
     text_enabled = True
 
+    fieldsets = [
+        (None, {
+            'fields': (
+                'picture',
+                'external_picture',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('width', 'height'),
+                'alignment',
+                'caption_text',
+                'attributes',
+            )
+        }),
+        (_('Link settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('link_url', 'link_page'),
+                'link_target',
+                'link_attributes',
+            )
+        }),
+        (_('Cropping settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('use_automatic_scaling', 'use_no_cropping'),
+                ('use_crop', 'use_upscale'),
+                'use_thumbnail',
+            )
+        })
+    ]
+
     def render(self, context, instance, placeholder):
-        if instance.url:
-            link = instance.url
-        elif instance.page_link:
-            link = instance.page_link.get_absolute_url()
-        else:
-            link = ""
+        classes = ''
+        if instance.alignment:
+            classes += 'align-{}'.format(instance.alignment)
+            if 'class' in instance.attributes:
+                classes += ' {}'.format(instance.attributes['class'])
+            # adapt new class
+            instance.attributes['class'] = classes
```

Comment:
I need to merge the alignment options with the class attribute if provided, so we end up with ``class="align-left my custom classes"`` instead of ``class="align-left" class="my custom classes"``